stetho-timber/src/main/java/com/facebook/stetho/timber/StethoTree.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.timber;
+
+import android.util.Log;
+import com.facebook.stetho.inspector.console.CLog;
+import com.facebook.stetho.inspector.network.NetworkPeerManager;
+import com.facebook.stetho.inspector.protocol.module.Console;
+import timber.log.Timber;
+
+public class StethoTree extends Timber.Tree {
+  @Override
+  protected void log(int priority, String tag, String message, Throwable t) {
+
+    NetworkPeerManager peerManager = NetworkPeerManager.getInstanceOrNull();
+    if (peerManager == null) {
+      return;
+    }
+
+    Console.MessageLevel logLevel;
+
+    switch (priority) {
+      case Log.VERBOSE:
+      case Log.DEBUG:
+        logLevel = Console.MessageLevel.DEBUG;
+        break;
+      case Log.INFO:
+        logLevel = Console.MessageLevel.LOG;
+        break;
+      case Log.WARN:
+        logLevel = Console.MessageLevel.WARNING;
+        break;
+      case Log.ERROR:
+        logLevel = Console.MessageLevel.ERROR;
+        break;
+      default:
+        logLevel = Console.MessageLevel.LOG;
+    }
+
+    CLog.writeToConsole(
```

Comment:
You've stumbled upon an unfortunate weakness in our API.  We don't really want to be using the NetworkPeerManager here (it's a subtle point admittedly but this may not be the set of peers who are observing console messages).

Instead, we want to make a new `ConsolePeerManager` which has a similar hacky API (`getInstanceOrNull`) and that is used by `Console` (peers are added/removed in `Console.enable`/`disable`.  Strictly speaking this means your log messages will be forwarded only to those peers which have invoked `Console.enable`, not necessarily those that called `Network.enable`.